### DISCUSSION TITLE
/bin/bash: line 604: cd: /opt/homebrew: Permission denied /bin/bash

### DISCUSSION THREAD

**Discussion by OKNoah** [Homebrew/discussions#836] [Getting started]
When doing

```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
```

I get

```
==> Downloading and installing Homebrew...
/bin/bash: line 604: cd: /opt/homebrew: Permission denied
/bin/bash: line 604: return: can only `return' from a function or sourced script
/usr/.git: Operation not permitted
Failed during: git init -q
```

This is on M1 Mac mostly fresh install expect for zinit and command line tools.

**Comment by SMillerDev** [comment]
What does `ls -al /opt/homebrew` or `ls -al /opt/*` show you?

**Reply by OKNoah** [reply]
```
naru@Narus-Pro ~ % ls -al /opt/homebrew
ls: /opt/homebrew: Permission denied
naru@Narus-Pro ~ % ls -al /opt/*
ls: /opt/homebrew: Permission denied
```

**Reply by OKNoah** [reply]
@SMillerDev I may have had `brew` installed before. I was separated from my computer for a few weeks so forgot what I'd installed to get up and running doing mobile dev. So I may have broken it.

**Reply by gromgit** [reply]
What's the output of `ls -ld@eO /opt/homebrew`?

**Reply by OKNoah** [reply]
> What's the output of `ls -ld@eO /opt/homebrew`?

```
naru@Narus-Pro ~ % ls -ld@eO /opt/homebrew
ls: /opt/homebrew: Permission denied
```

**Reply by gromgit** [reply]
Huh. How about `ls -ld@eO /opt`?

**Reply by OKNoah** [reply]
> Huh. How about `ls -ld@eO /opt`?

```
drw-r--r--  3 naru  wheel  hidden 96 Feb 16 22:09 /opt
```

**Reply by gromgit** [reply]
That's not right at all. Did you happen to do a `chmod -R 644 /opt` or something similar at some point?

Anyway, to fix this, do the following:
```
find /opt -type d -exec chmod a+x {} +
```
then run `/opt/homebrew/bin/brew config` and `/opt/homebrew/bin/brew doctor` to see if you now have a working Homebrew setup.

**Reply by OKNoah** [reply]
> That's not right at all. Did you happen to do a `chmod -R 644 /opt` or something similar at some point?
> 
> Anyway, to fix this, do the following:
> 
> ```
> find /opt -type d -exec chmod a+x {} +
> ```
> 
> then run `/opt/homebrew/bin/brew config` and `/opt/homebrew/bin/brew doctor` to see if you now have a working Homebrew setup.

thanks very much

```
naru@Narus-Pro ~ % /opt/homebrew/bin/brew config
HOMEBREW_VERSION: 3.0.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: c951be8d3c7b339c7e759b7c40aec859e09a70a5
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a3f5050b1321d621cd392842d1cede81637124a9
Core tap last commit: 46 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.1-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: 12.4
Rosetta 2: false
naru@Narus-Pro ~ % /opt/homebrew/bin/brew doctor
Your system is ready to brew.
```

**Comment by saioom-sahoo** [comment]
find /opt -type d -exec chmod a+x {} +
find: /opt: Permission denied

**Reply by gromgit** [reply]
That error says `find` can't even read `/opt`. What's the output of `/bin/ls -lde@ /opt`?
